File: dm-llib/src/main/java/com/ysz/dm/lib/validator/HibernateValidateTools.java
package com.ysz.dm.lib.validator;

import java.util.Objects;
import java.util.Set;
import javax.validation.ConstraintViolation;
import javax.validation.Validation;
import javax.validation.Validator;
import javax.validation.ValidatorFactory;
import org.hibernate.validator.HibernateValidator;

/**
 * <pre>
 * class desc here
 * </pre>
 *
 * @author carl.yu
 * @createAt 2022/11/9
 **/
public class HibernateValidateTools {

  private static ValidatorFactory factory;
  private static Validator validator;

  static {
    factory =
        Validation.byProvider(HibernateValidator.class)
            .configure()
            // 快速失败模式
            .failFast(true)
            // .addProperty( "hibernate.validator.fail_fast", "true" )
            .buildValidatorFactory();
    validator = factory.getValidator();
  }


  public static void chkAndThrow(Object bean) throws RuntimeException {
    validator = factory.getValidator();
    Set<ConstraintViolation<Object>> validateRes = validator.validate(bean);
//    if (CollectionTools.isNotEmpty(validateRes)) {
    if (null != validateRes && validateRes.size() > 0) {

      ConstraintViolation<Object> first = validateRes.iterator().next();
      if (first != null) {
        String path = Objects.toString(first.getPropertyPath());
        String message = first.getMessage();
        throw new IllegalArgumentException(path + ":" + message);
      }
    }
  }


}

File: rb/rb-base-core/src/main/kotlin/com/ysz/dm/rb/base/core/tools/hibernate/HibernateValidateTools.kt
package com.ysz.dm.rb.base.core.tools.hibernate

import com.ysz.dm.rb.base.core.exceptions.BaseRbParamException
import org.hibernate.validator.HibernateValidator
import java.util.*
import javax.validation.Validation
import javax.validation.ValidatorFactory

/**
 * @author carl
 */
class HibernateValidateTools {

    companion object {
        private val factory: ValidatorFactory = Validation.byProvider(
            HibernateValidator::class.java
        ).configure() // 快速失败模式
            .failFast(true) // .addProperty( "hibernate.validator.fail_fast", "true" )
            .buildValidatorFactory()
        private val validator = factory.validator

        @Throws(BaseRbParamException::class)
        fun chkThenThrow(bean: Any) {
            chkThenThrow(bean, false)
        }

        @Throws(BaseRbParamException::class)
        fun chkThenThrow(bean: Any, messageWithPath: Boolean) {
            val validateRes = validator!!.validate(bean)
            if (!validateRes.isNullOrEmpty()) {
                val first = validateRes.iterator().next()
                if (first != null) {
                    val path = Objects.toString(first.propertyPath)
                    val message = first.message
                    throw BaseRbParamException(
                        if (messageWithPath) "$path:$message" else "$message"
                    )
                }
            }
        }
    }
}